service/src/main/java/com/commercetools/pspadapter/payone/IntegrationService.java

```diff
@@ -32,51 +27,74 @@
 
     public static final String HEROKU_ASSIGNED_PORT = "PORT";
 
-    private final CommercetoolsQueryExecutor commercetoolsQueryExecutor;
-    private final PaymentDispatcher paymentDispatcher;
-    private final CustomTypeBuilder typeBuilder;
-    private final NotificationDispatcher notificationDispatcher;
-    private final String payoneInterfaceName;
-
-    IntegrationService(
-            final CustomTypeBuilder typeBuilder,
-            final CommercetoolsQueryExecutor commercetoolsQueryExecutor,
-            final PaymentDispatcher paymentDispatcher,
-            final NotificationDispatcher notificationDispatcher,
-            final String payoneInterfaceName) {
-        this.commercetoolsQueryExecutor = commercetoolsQueryExecutor;
-        this.paymentDispatcher = paymentDispatcher;
-        this.typeBuilder = typeBuilder;
-        this.notificationDispatcher = notificationDispatcher;
-        this.payoneInterfaceName = payoneInterfaceName;
+    private final List<TenantFactory> tenantFactories;
+
+    public IntegrationService(final List<TenantFactory> tenantFactories) {
+        if (CollectionUtils.isEmpty(tenantFactories)) {
+            throw new IllegalArgumentException("Tenants list must be non-empty");
+        }
+
+        this.tenantFactories = tenantFactories;
+    }
+
+    /**
+     * @return Unmodifiable view of tenant factories list which are used for the service run.
+     */
+    public List<TenantFactory> getTenantFactories() {
+        return Collections.unmodifiableList(tenantFactories);
     }
 
     public void start() {
-        createCustomTypes();
 
+        initSparkService();
+
+        for (TenantFactory tenantFactory : tenantFactories) {
+            initTenantServiceResources(tenantFactory);
+        }
+
+        Spark.awaitInitialization();
+    }
+
+    private void initSparkService() {
         Spark.port(port());
 
         // This is a temporary jerry-rig for the load balancer to check connection with the service itself.
         // For now it just returns a JSON response {"status":200}
         // It should be expanded to a more real health-checker service, which really performs PAYONE status check.
         // But don't forget, a load balancer may call this URL very often (like 1 per sec),
         // so don't make this request processor heavy, or implement is as independent background service.
+        LOG.info("Register /health URL");
         Spark.get("/health", (req, res) -> {
             res.status(HttpStatusCode.OK_200);
             res.type(ContentType.APPLICATION_JSON.getMimeType());
             return ImmutableMap.of("status", HttpStatus.OK_200);
         }, SphereJsonUtils::toJsonString);
+    }
+
+    private static void initTenantServiceResources(TenantFactory tenantFactory) {
+
+        // create custom types
+        tenantFactory.getCustomTypeBuilder().run();
+
+        PaymentHandler paymentHandler = tenantFactory.getPaymentHandler();
```

Comment:
Why? it takes:
  - more time
  - more space
  - redundant re-factoring hundreds of lines
  - a little profit

I'm agree the `final` sometimes is useful, but i don't want to re-factor half of the project because of this convention now.